pyvisa-py/highlevel.py

```diff
@@ -431,9 +470,146 @@ def unlock(self, session):
         return sess.unlock()
 
     def disable_event(self, session, event_type, mechanism):
-        # TODO: implement this for GPIB finalization
-        pass
+        """Disables notification of the specified event type(s) via the specified mechanism(s).
+
+        Corresponds to viDisableEvent function of the VISA library.
+
+        :param session: Unique logical identifier to a session.
+        :param event_type: Logical event identifier.
+        :param mechanism: Specifies event handling mechanisms to be disabled.
+                        (Constants.VI_QUEUE, .VI_HNDLR, .VI_SUSPEND_HNDLR, .VI_ALL_MECH)
+        :return: return value of the library call.
+        :rtype: :class:`pyvisa.constants.StatusCode`
+        """
+        try:
+            sess = self.sessions[session]
+        except KeyError:
+            return StatusCode.error_invalid_object
+
+        # TODO: add support for VI_HNDLR, VI_SUSPEND_HNDLR, VI_ALL_MECH
```

Comment:
I feel like the session should filter itself for what kind of event it supports like you already did in GPIB.

Reply:
Before factoring out common functionality to the Session parent class, I am inclined to see some other Session types support events. Do you think different Session subclasses would need fancier filtering, eg. by mechanism and event type?

Reply:
I think there is a misunderstanding here. My point is that the VisaLibrary should not filter the mechanisms but let the session decide if it supports the mechanism.

Reply:
Ah, you're correct, sorry. I agree of course, the session should filter itself.

Here I was quite confused about separation of concerns. For instance, I see that pyvisa.VisaLibraryBase already defines a .handlers member instead of having handler registration, storage, mechanism etc. delegated to derived classes.

Reply:
I guess that it is because pyvisa when using NI-VISA (or similar) does have access to sessions but need to keep track of handlers.

Reply:
Looks that way, yes. I'd say keeping track of handlers is an implementation detail that should go into NIVisaLibrary.